modules/cli/src/main/scala/scala/cli/commands/ScalaCommand.scala

```diff
@@ -202,4 +202,56 @@ abstract class ScalaCommand[T](implicit myParser: Parser[T], help: Help[T])
           // This requires writing our own minimal JNI library, that publishes '.a' files too for static linking in the executable of Scala CLI.
           None
       }
+
+  /** @param options
+    *   command-specific [[T]] options
+    * @return
+    *   Tries to create BuildOptions based on [[sharedOptions]] and exits on error. Override to
+    *   change this behaviour.
+    */
+  def buildOptions(options: T): Option[BuildOptions] =
+    sharedOptions(options).map(shared => shared.buildOptions().orExit(shared.logger))
+
+  protected def buildOptionsOrExit(options: T): BuildOptions =
+    buildOptions(options) match {
+      case Some(bo) => bo
+      case _ =>
+        sharedOptions(options).foreach(_.logger.debug("build options could not be initialized"))
+        sys.exit(1)
+    }
+
+  /** @param options
+    *   command-specific [[T]] options
+    * @return
+    *   by default [[sharedOptions]].logging, override to adjust.
+    */
+  def loggingOptions(options: T): Option[LoggingOptions] =
+    sharedOptions(options).map(_.logging)
+
+  /** @param options
+    *   command-specific [[T]] options
+    * @return
+    *   by default [[loggingOptions]].verbosity, override to adjust.
+    */
+  def verbosity(options: T): Option[Int] =
+    loggingOptions(options).map(_.verbosity)
+
+  /** This should be overridden instead of [[run]] when extending [[ScalaCommand]].
+    *
+    * @param options
+    *   the command's specific set of options
+    * @param remainingArgs
+    *   arguments remaining after parsing options
+    */
+  def scalaCliRun(options: T, remainingArgs: RemainingArgs): Unit
```

Comment:
```suggestion
  def runCommand(options: T, remainingArgs: RemainingArgs): Unit
```

What do you think about changing name of this function to `runCommand`?

Reply:
actually, that's not bad. Lemme rename it.

Reply:
done